### DISCUSSION TITLE
How to get text from label?

### DISCUSSION THREAD

**Discussion by Harsha-Gelato** [appium/appium#17548] [General]
Hi All,

I am using appium flutter driver with wdio and there is one element in heirachy from which I want to extract text for the price and verify it
<img width="488" alt="Screenshot 2022-10-03 at 12 06 11" src="https://user-images.githubusercontent.com/103107117/193598267-8569c6b4-d7fa-4dd1-9a8d-d50e9cfa3749.png">

Below is the dev code for the same:

   return Semantics(
          label: text.stripHtml,
          child: Wrap(crossAxisAlignment: WrapCrossAlignment.center, children: widgets,),
          key: Key("catalog_product_card_price")
        );

Below is the script code to get text from it:

> for (const catalogItem of menClothingSubCatalogMap[subCatalog]) {
>         log.info(`Navigating to ${catalogItem} ${`catalog_product_#${index}`}`);
>         await driver.execute('flutter:scrollIntoView', find.byValueKey(`catalog_product_#${index}`), { alignment: 0.1 });
>         await driver.execute('flutter:waitFor', find.byValueKey(`catalog_product_#${index}`));
>         const catalogPrice = find.descendant({
>             of: find.byValueKey(`catalog_product_#${index}`),
>             matching: find.bySemanticsLabel('catalog_product_card_price'),
>         });
>         console.log('****************');
>         console.log(catalogPrice);
> 
>         const price =  await driver.getElementText(catalogPrice);
>         const regex = /^[¢£$kc¥€]{1}\d{1,}(\.\d{1,3})?$/ig;
>         const results = regex.test(price);
>         expect(results).to.be.true;
>         index++;
>     }

In my case **getElementText()** is not working using above approach.
Can someone pls tell me what I am missing at my end?
Thanks :slight_smile:
Below is the console log for the value of **catalogPrice**
<img width="1099" alt="Screenshot 2022-10-03 at 14 16 09" src="https://user-images.githubusercontent.com/103107117/193598393-cc254146-a39c-4627-9236-782554dccf39.png">

Below is the error I am facing using **getElementText()**
<img width="1104" alt="Screenshot 2022-10-03 at 14 14 44" src="https://user-images.githubusercontent.com/103107117/193598355-5b3381db-635d-4b00-9f04-fe6fef6b8ec9.png">

**Comment by Mr-Ples** [comment]
Try to get the name of `catalogPrice` and check that you are getting the element you are expecting to get. Maybe the `catalogPrice` returns the wrong element, one that has no text?

**Comment by Mr-Ples** [comment]
Also try adding `byValueKey`:

```
await driver.getElementText(find.byValueKey(catalogPrice))
```
